### PR TITLE
[Merged by Bors] - feat: port Data.Set.Intervals.Disjoint

### PR DESCRIPTION
**Pull request by riccardobrasca**

### CODE REVIEW

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+theorem IsLUB.bunionᵢ_Iic_eq_Iio (a_lub : IsLUB s a) (a_not_mem : a ∉ s) :
+    (⋃ x ∈ s, Iic x) = Iio a :=
+  a_lub.dual.bunionᵢ_Ici_eq_Ioi a_not_mem
+#align is_lub.bUnion_Iic_eq_Iio IsLUB.bunionᵢ_Iic_eq_Iio
+
+theorem IsLUB.bunionᵢ_Iic_eq_Iic (a_lub : IsLUB s a) (a_mem : a ∈ s) : (⋃ x ∈ s, Iic x) = Iic a :=
+  a_lub.dual.bunionᵢ_Ici_eq_Ici a_mem
+#align is_lub.bUnion_Iic_eq_Iic IsLUB.bunionᵢ_Iic_eq_Iic
+
+theorem unionᵢ_Ici_eq_Ioi_infi {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:230]:
```suggestion
theorem unionᵢ_Ici_eq_Ioi_infᵢ {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    (no_least_elem : (⨅ i, f i) ∉ range f) : (⋃ i : ι, Ici (f i)) = Ioi (⨅ i, f i) := by
+  simp only [← IsGLB.bunionᵢ_Ici_eq_Ioi (@isGLB_infᵢ _ _ _ f) no_least_elem, mem_range,
+    unionᵢ_exists, unionᵢ_unionᵢ_eq']
+#align Union_Ici_eq_Ioi_infi unionᵢ_Ici_eq_Ioi_infi
+
+theorem unionᵢ_Iic_eq_Iio_supr {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:236]:
```suggestion
theorem unionᵢ_Iic_eq_Iio_supᵢ {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    (no_greatest_elem : (⨆ i, f i) ∉ range f) : (⋃ i : ι, Iic (f i)) = Iio (⨆ i, f i) :=
+  @unionᵢ_Ici_eq_Ioi_infi ι (OrderDual R) _ f no_greatest_elem
+#align Union_Iic_eq_Iio_supr unionᵢ_Iic_eq_Iio_supr
+
+theorem unionᵢ_Ici_eq_Ici_infi {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:241]:
```suggestion
theorem unionᵢ_Ici_eq_Ici_infᵢ {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```

File: Mathlib/Data/Set/Intervals/Disjoint.lean [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    (has_least_elem : (⨅ i, f i) ∈ range f) : (⋃ i : ι, Ici (f i)) = Ici (⨅ i, f i) := by
+  simp only [← IsGLB.bunionᵢ_Ici_eq_Ici (@isGLB_infᵢ _ _ _ f) has_least_elem, mem_range,
+    unionᵢ_exists, unionᵢ_unionᵢ_eq']
+#align Union_Ici_eq_Ici_infi unionᵢ_Ici_eq_Ici_infi
+
+theorem unionᵢ_Iic_eq_Iic_supr {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Disjoint.lean:247]:
```suggestion
theorem unionᵢ_Iic_eq_Iic_supᵢ {R : Type _} [CompleteLinearOrder R] {f : ι → R}
```

**Review by dupuisf — COMMENTED** [review]
Just a few lemma names to fix, but otherwise LGTM!

bors d+

**Comment by riccardobrasca** [comment]
bors merge

**Pull request closed without merging**